### PR TITLE
Safer way to handle secret information of cli_wallet #1171

### PR DESCRIPTION
**Pull request by cogutvalera**
This PR is for fixing https://github.com/bitshares/bitshares-core/issues/1171 issue

**Safer way to handle secret information of cli_wallet**

Depends on already merged PRs:
1. bitshares/bitshares-fc#86
2. bitshares/bitshares-fc#82

### CODE REVIEW

File: src/rpc/cli.cpp [hunk 1]
```diff
@@ -250,6 +249,12 @@ void cli::getline( const fc::string& prompt, fc::string& line)
             FC_THROW_EXCEPTION( fc::eof_exception, "" );
          line = line_read;
          // we don't need here to add line in editline's history, cause it will be doubled
+         if (cli_check_secret(line_read)) {
+            el_no_echo = 1;
+            line_read = readline("Enter password: ");
```
Comment (pmconrad) [inline src/rpc/cli.cpp]: Must `free(line_read)` before re-using it

File: src/rpc/cli.cpp [hunk 2]
```diff
@@ -250,6 +249,12 @@ void cli::getline( const fc::string& prompt, fc::string& line)
...
+            el_no_echo = 0;
+            line = line + ' ' + line_read;
```
Comment (pmconrad) [inline src/rpc/cli.cpp]: Must add check for `null` like above

**Review by pmconrad — CHANGES_REQUESTED** [review]
I like this approach much better.

**Comment by abitmore** [comment]
@cogutvalera do you have time to continue the work on this?

**Comment by cogutvalera** [comment]
> @cogutvalera do you have time to continue the work on this?

yes sure

**Comment by pmconrad** [comment]
> yes sure

@cogutvalera any estimate when you do this?

**Comment by cogutvalera** [comment]
I will try to do it till the end of this week, was unplanned busy

File: src/rpc/cli.cpp (hunk 1 above)
Comment (cogutvalera) [reply src/rpc/cli.cpp]: done

File: src/rpc/cli.cpp (hunk 2 above)
Comment (cogutvalera) [reply src/rpc/cli.cpp]: done

**Review by pmconrad — APPROVED** [review]
Thanks!

**Pull request merged**